### PR TITLE
chore(python): add a `make requirements` option to install/refresh dependencies without having to recreate the `venv`

### PR DESCRIPTION
**Pull request by alexander-beedie**
Inspired by a Discord [thread](https://discord.com/channels/908022250106667068/908022461751234570/1089265752550416385), I've added...
```
make requirements
```
...which will update/install everything listed in our various `requirements` files _without_ having to delete/recreate the associated `venv`. Slightly less friction staying up to date as dependencies advance.

### CODE REVIEW

**Comment by alexander-beedie** [comment]
@stinodego I think you're also tweaking the Makefile? 
Will wait until you merge to rebase against your changes (if needed) 😎👍

**Review by stinodego — DISMISSED** [review]
This is great - it's the option that was missing :)

File: py-polars/Makefile
Comment (stinodego) [inline py-polars/Makefile:18]: Actually, spotted one thing: `requirements` depends on `venv`. I tested this and it doesnt end up in an infinite loop or anything.

```suggestion
requirements: venv  ## Install/refresh all project requirements
```

Comment (alexander-beedie) [reply py-polars/Makefile]: Integrated 👍

Comment (stinodego) [reply py-polars/Makefile:18]: In the end, it doesn't matter too much. But don't you think it's nicer that it will set up the `venv` for you, rather than error out?

Comment (alexander-beedie) [reply py-polars/Makefile]: True... why not indeed :)

**Pull request merged**
